src/gt4py/next/program_processors/runners/dace_fieldview/transformations/local_double_buffering.py

```diff
@@ -0,0 +1,393 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2024, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+
+import copy
+
+import dace
+from dace import (
+    data as dace_data,
+    dtypes as dace_dtypes,
+    symbolic as dace_symbolic,
+    transformation as dace_transformation,
+)
+from dace.sdfg import nodes as dace_nodes
+
+from gt4py.next.program_processors.runners.dace_fieldview import (
+    transformations as gtx_transformations,
+)
+
+
+def gt_crearte_local_double_buffering(
+    sdfg: dace.SDFG,
+) -> int:
+    """Modifies the SDFG that point wise data dependencies are stable.
+
+    Rule 3 of the ADR18, guarantees that if data is input and output to a map,
+    then it must be a non transient array and it must only have point wise
+    dependency. This means that every index that is read is also written by
+    the same thread and no other thread reads or writes to the same location.
+    However, because the dataflow inside a map is partially asynchronous
```

Comment:
I agree with what is written here. I think we also agree that this function is needed because we "aggressively" remove double buffers on transient arrays, that were there in the original SDFG. Let's say that this is our custom approach, but I think it will be hard to generalize it.